python/service/service.py

```diff
@@ -331,6 +331,73 @@ def handle_pnp_command_invoke(self, device_data, event):
             )
         )
 
+    def update_pairing(self, device_id, thief):
+        # Handle the case where a device is looking for someone to pair with
+        if thief.get(Fields.CMD, "") == Commands.PAIR_WITH_SERVICE_APP:
+            if thief.get(Fields.REQUESTED_SERVICE_POOL, None) == service_pool:
+                received_operation_id = thief.get(Fields.OPERATION_ID)
+                received_run_id = thief.get(Fields.RUN_ID)
+
+                logger.info(
+                    "Device {} attempting to pair with operation ID {}".format(
+                        device_id, received_operation_id
+                    ),
+                    extra=custom_props(device_id, received_run_id),
+                )
+
+                message = json.dumps(
+                    {
+                        Fields.THIEF: {
+                            Fields.CMD: Commands.PAIR_RESPONSE,
+                            Fields.SERVICE_INSTANCE_ID: service_instance_id,
+                            Fields.RUN_ID: received_run_id,
+                            Fields.OPERATION_ID: received_operation_id,
+                        }
+                    }
+                )
+
+                self.outgoing_c2d_queue.put(
+                    OutgoingC2d(
+                        device_id=device_id, message=message, props=Const.JSON_TYPE_AND_ENCODING,
+                    )
+                )
+
+                # don't add device_id to self.device_list until we get a message with our
+                # service_instance_id
+
+        elif Fields.RUN_ID in thief and Fields.SERVICE_INSTANCE_ID in thief:
+            received_service_instance_id = thief.get(Fields.SERVICE_INSTANCE_ID, None)
+            received_run_id = thief[Fields.RUN_ID]
+
+            device_data = self.device_list.try_get(device_id)
+
+            # Handle the case where a device pairs with us
+            if not device_data and received_service_instance_id == service_instance_id:
+                device_data = PerDeviceData(device_id, received_run_id)
+                self.device_list.add(device_id, device_data)
+
+            # Handle the case where we used to be paired and we're not anymore
+            elif device_data and not received_service_instance_id == service_instance_id:
```

Comment:
nit:
```suggestion
            elif device_data and received_service_instance_id != service_instance_id:
```

Reply:
I agree. but the Python linter prefers the way it was in my PR and it's just not worth arguing with the linter....